tests/unit/s2n_tls13_new_session_ticket_test.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "testlib/s2n_testlib.h"
+
+#include <s2n.h>
+
+#include "tls/s2n_connection.h"
+#include "tls/s2n_tls13.h"
+#include "tls/s2n_record.h"
+#include "tls/s2n_post_handshake.h"
+#include "tls/s2n_tls13_handshake.h"
+
+#define MAX_TEST_SESSION_SIZE 300
+
+uint8_t session_ticket_counter = 0;
+static int s2n_test_session_ticket_cb(struct s2n_connection *conn, struct s2n_session_ticket *ticket)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(ticket);
+
+    size_t cb_session_data_len = 0;
+    uint8_t cb_session_data[MAX_TEST_SESSION_SIZE] = { 0 };
+    uint32_t cb_session_lifetime = 0;
+
+    EXPECT_SUCCESS(s2n_session_ticket_get_data_len(ticket, &cb_session_data_len));
+    EXPECT_TRUE(cb_session_data_len > 0);
+    EXPECT_SUCCESS(s2n_session_ticket_get_data(ticket, cb_session_data_len, cb_session_data));
+    EXPECT_NOT_NULL(cb_session_data);
+    EXPECT_SUCCESS(s2n_session_ticket_get_lifetime(ticket, &cb_session_lifetime));
+    EXPECT_TRUE(cb_session_lifetime > 0);
+
+    session_ticket_counter++;
+
+    return S2N_SUCCESS;
+}
+
+static int s2n_setup_test_ticket_key(struct s2n_config *config)
+{
+    POSIX_ENSURE_REF(config);
+
+    /**
+     *= https://tools.ietf.org/rfc/rfc5869#appendix-A.1
+     *# PRK  = 0x077709362c2e32df0ddc3f0dc47bba63
+     *#        90b6c73bb50f9c3122ec844ad7c2b3e5 (32 octets)
+     **/
+    S2N_BLOB_FROM_HEX(ticket_key,
+    "077709362c2e32df0ddc3f0dc47bba63"
+    "90b6c73bb50f9c3122ec844ad7c2b3e5");
+
+    /* Set up encryption key */
+    uint64_t current_time;
+    uint8_t ticket_key_name[16] = "2016.07.26.15\0";
```

Comment:
Nit:  
Zero initialize here:`uint64_t current_time  = 0;`. Also, you don't have to mention explicitly size `uint8_t ticket_key_name[] = "2016.07.26.15\0";` should be good enough here :)

Reply:
Turns out key names need to be explicitly 16 bytes, if I don't add a length I get a funky buffer overflow error. I could probably increase the string to be exactly 16 bytes instead, but I just decided to use an explicit length.